src/parser/util.js

```diff
@@ -32,57 +35,57 @@ export default class UtilParser extends Tokenizer {
 
   // Tests whether parsed token is a contextual keyword.
 
-  isContextual(name) {
+  isContextual(name: string): boolean {
     return this.match(tt.name) && this.state.value === name;
   }
 
   // Consumes contextual keyword if possible.
 
-  eatContextual(name) {
+  eatContextual(name: string): boolean {
     return this.state.value === name && this.eat(tt.name);
   }
 
   // Asserts that following token is given contextual keyword.
 
-  expectContextual(name, message) {
+  expectContextual(name: string, message?: string): void {
     if (!this.eatContextual(name)) this.unexpected(null, message);
   }
 
   // Test whether a semicolon can be inserted at the current position.
 
-  canInsertSemicolon() {
+  canInsertSemicolon(): boolean {
     return this.match(tt.eof) ||
       this.match(tt.braceR) ||
       lineBreak.test(this.input.slice(this.state.lastTokEnd, this.state.start));
   }
 
   // TODO
 
-  isLineTerminator() {
+  isLineTerminator(): boolean {
     return this.eat(tt.semi) || this.canInsertSemicolon();
   }
 
   // Consume a semicolon, or, failing that, see if we are allowed to
   // pretend that there is a semicolon at this position.
 
-  semicolon() {
+  semicolon(): void {
```

Comment:
Without the return type, you should still get errors if there are errors, but they will be more confusing: If you have `function f() { return g(); }` and call `f().toUpperCase()`, and there's an error, is it in `f`, `g`, or `.toUpperCase()`?